### PR TITLE
feat: refactor Camera and ImagePicker as injected dependency

### PR DESCRIPTION
**Pull request by loc-nguyenthien**

### CODE REVIEW

**Review by huydosgtech — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -11,92 +11,136 @@ npm install rn-firebase-chat
...
+      hasGallery={true}
+      inputToolbarProps={{
+        onPressCamera,
+        onPressGallery,
+      }}
+    >
+    {({onSend, userInfo}) => (
+      <CameraView
+        onSend={onSend}
+        userInfo={userInfo}
```
Comment (loc-nguyenthien) [inline README.md:135]: I think we already moved this props into `CameraView` component

File: package.json [hunk 2]
```diff
@@ -8,7 +8,8 @@
   "source": "src/index",
   "files": [
     "src",
-    "lib"
+    "lib",
+    "addons"
```
Comment (loc-nguyenthien) [inline package.json:12]: remove this, since we already moved `addons` into `src`

File: tsconfig.json [hunk 3]
```diff
@@ -24,5 +24,5 @@
     "strict": true,
     "target": "esnext"
   },
-  "include": ["src/**/*"]
+  "include": ["src/**/*", "addons/**/*"]
```
Comment (loc-nguyenthien) [inline tsconfig.json:27]: remove this, since we already moved `addons` into `src`

File: README.md (hunk 1 above)
Comment (hoangSTS) [reply README.md]: updated

File: package.json (hunk 2 above)
Comment (hoangSTS) [reply package.json]: updated

File: tsconfig.json (hunk 3 above)
Comment (hoangSTS) [reply tsconfig.json]: updated

**Comment by loc-nguyenthien** [comment]
Approved
(Cannot approve my own PR)

File: src/chat/ChatScreen.tsx [hunk 4]
```diff
@@ -69,7 +67,7 @@ interface ChatScreenProps extends GiftedChatProps {
   typingTimeoutSeconds?: number;
   messageStatusEnable?: boolean;
   customMessageStatus?: (hasUnread: boolean) => JSX.Element;
-  iconsCamera?: IconPaths;
+  children?: (props: ChildrenProps) => ReactNode | ReactNode;
```
Comment (huydosgtech) [inline src/chat/ChatScreen.tsx:70]: seem we miss action onPressCamera. Can you check it?

File: README.md [hunk 5]
```diff
@@ -11,92 +11,131 @@ npm install rn-firebase-chat
...
+  const {onPressCamera, onPressGallery} = useCamera()
   return (
     <BaseChatScreen
       memberIds={[partnerInfo.id]}
       partners={[partnerInfo]}
-    />
+      hasCamera={true}
+      hasGallery={true}
+      inputToolbarProps={{
+        onPressCamera,
```
Comment (huydosgtech) [inline README.md:128]: Do you think we should move hasCamera, hasGallery into inputToolbarProps?

**Review by huydosgtech — CHANGES_REQUESTED** [review]

File: src/chat/ChatScreen.tsx (hunk 4 above)
Comment (hoangSTS) [reply src/chat/ChatScreen.tsx:67]: removed

File: README.md (hunk 5 above)
Comment (hoangSTS) [reply README.md:128]: moved

**Review by honghoangsts — APPROVED** [review]

**Pull request merged**